tests/Integration/Core/Translation/Storage/Provider/CoreDomainCatalogueLayersProviderTest.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Integration\Core\Translation\Storage\Provider;
+
+use PrestaShop\PrestaShop\Core\Language\LanguageRepositoryInterface;
+use PrestaShop\PrestaShop\Core\Translation\Storage\Provider\CoreCatalogueLayersProvider;
+use PrestaShop\PrestaShop\Core\Translation\Storage\Provider\Definition\CoreDomainProviderDefinition;
+use PrestaShop\PrestaShop\Core\Translation\TranslationRepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Test the core translations provider filtering by domain
+ */
+class CoreDomainCatalogueLayersProviderTest extends KernelTestCase
+{
+    /**
+     * @var string
+     */
+    protected $translationsDir;
+
+    protected function setUp()
+    {
+        self::bootKernel();
+        $this->translationsDir = self::$kernel->getContainer()->getParameter('test_translations_dir');
+    }
+
+    /**
+     * Test it loads a XLIFF catalogue from the locale's `translations` directory
+     */
+    public function testItLoadsCatalogueFromXliffFilesInLocaleDirectory(): void
```

Comment:
Consider using a dataProvider with two data sets, one for AdminActions and another or ModulesCheckpaymentAdmin